lib/db.js

```diff
@@ -419,19 +450,38 @@ module.exports = (
 
   DB.prototype.updateSessionToken = function (token, userAgentString) {
     log.trace({ op: 'DB.updateSessionToken', uid: token && token.uid })
-
-    if (! token.update(userAgentString)) {
+    if (! token.update(userAgentString) || ! this.redis) {
       return P.resolve()
     }
 
-    return this.pool.post('/sessionToken/' + token.id + '/update', {
+    const uid = token.uid
+    const newToken = [{
+      tokenId: token.id,
+      uid: uid,
       uaBrowser: token.uaBrowser,
       uaBrowserVersion: token.uaBrowserVersion,
       uaOS: token.uaOS,
       uaOSVersion: token.uaOSVersion,
       uaDeviceType: token.uaDeviceType,
-      lastAccessTime: token.lastAccessTime
-    })
+      lastAccessTime: token.lastAccessTime,
+      createdAt: token.createdAt
+    }]
+    let sessionTokens = []
+    if (features.isLastAccessTimeEnabledForUser(uid)) {
+      return this.redis.getAsync(uid)
+      .then(res => {
+        if (res) {
+          res = JSON.parse(res)
+          const filteredRes = res.filter(tok => tok.tokenId !== token.id)
```

Comment:
Could you please add a short doc string that explains what the several lines below are doing. This is for future readers of this code...